### ISSUE TITLE
CUDA capability not detected correctly with --try-toolchain

### ISSUE THREAD

**Issue by mboisson** [easybuilders/easybuild-framework#2620]
We noticed that the new `--try-toolchain` behavior breaks our MNS. More specifically, when specifying a cuda toolchain, it no longer installs under the CUDA directory. 

For example : 
```
$ eb OpenMPI-2.1.1-iccifortcuda-2016.4.eb --try-toolchain=iccifortcuda,2016.4.100
== temporary log file in case of crash /tmp/eb-_nmzG4/easybuild-_8drJC.log
== avx2/Compiler/intel2016.4/openmpi/2.1.1 is already installed (module found), skipping
== No easyconfigs left to be built.
== Build succeeded for 0 out of 0
== Temporary log file(s) /tmp/eb-_nmzG4/easybuild-_8drJC.log* have been removed.
== Temporary directory /tmp/eb-_nmzG4 has been removed.
``` 

Previously, it would have installed under `avx2/CUDA/intel2016.4/cuda10.0/openmpi/2.1.1`. Adding some debugging code into our MNS (namely, I added `print("ec: %s" % str(ec.toolchain.definition()))` in `det_module_subdir`, I see that `det_module_subdir` is not being called with consistent toolchain descriptions throughout : 
```
[mboisson@build-node OpenMPI]$ ~/myeb --debug OpenMPI-2.1.1-iccifortcuda-2016.4.eb --try-toolchain=iccifortcuda,2016.4.100
== temporary log file in case of crash /tmp/eb-gPWJ9O/easybuild-519I3R.log
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER_CUDA': ['CUDA'], 'COMPILER': ['icc', 'ifort', 'CUDA']}
ec: {'COMPILER_CUDA': ['CUDA'], 'COMPILER': ['icc', 'ifort', 'CUDA']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER_CUDA': ['CUDA'], 'COMPILER': ['icc', 'ifort', 'CUDA']}
ec: {'COMPILER_CUDA': ['CUDA'], 'COMPILER': ['icc', 'ifort', 'CUDA']}
ec: {'COMPILER': ['icc', 'ifort']}
ec: {'COMPILER': ['icc', 'ifort']}
== avx2/Compiler/intel2016.4/openmpi/2.1.1 is already installed (module found), skipping
``` 

@bartoldeman @ocaisa

**Comment by mboisson** [comment]
For reference, our MNS is here : https://github.com/ComputeCanada/easybuild-framework/blob/computecanada-master/easybuild/tools/module_naming_scheme/SoftCCHierarchicalMNS.py

**Comment by mboisson** [comment]
More progress. Problem identified in `toolchain_mapping: {'iccifortcuda': {'version': '7.3.0', 'name': 'GCC'}, 'iccifort': {'version': '7.3.0', 'name': 'GCC'}, 'GCCcore': {'version': '7.3.0', 'name': 'GCCcore'}}`  in `tweak.py`

It maps `iccifortcuda` to `GCC` because the toolchain capability detection reports `'cuda': None`, in easyconfig.py at

```
 287                 if capability == TOOLCHAIN_CAPABILITY_CUDA:
 288                     # use None rather than False, useful to have it consistent with the rest
 289                     toolchain[capability] = ('CUDA_CC' in tc.variables) or None
```

**Comment by mboisson** [comment]
Renamed the issue since the source seems to be that CUDA capability detection is broken.

**Comment by mboisson** [comment]
I am not sure what `tc.variables` was supposed to be, but in our case, that was an empty dictionary. I suggest instead testing if the class is an instance of the `Cuda` toolchain. PR #2621 is doing just that.

**Comment by boegel** [comment]
fixed in #2621, thanks @mboisson!
